llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -218,6 +218,16 @@ Instruction *InstCombinerImpl::visitMul(BinaryOperator &I) {
                   : BinaryOperator::CreateNeg(Op0);
   }
 
+  {
+    // abs(a * abs(b)) --> abs(a * b)
```

Comment:
Shouldn't we implement the simplification in `InstCombinerImpl::visitCallInst`?

Reply:
Yes, it should. Pretty sure the original patch did do it that way, why was it moved here? Now it's folding `a * abs(b)` to `a * b`, which isn't right.

Reply:
@dtcxzyw, @nikic updated.